src/control/controlproxyqml.cpp

```diff
@@ -0,0 +1,68 @@
+#include "control/controlproxyqml.h"
+
+#include "moc_controlproxyqml.cpp"
+
+ControlProxyQml::ControlProxyQml(QObject* parent)
+        : QObject(parent),
+          m_pControlProxy(nullptr) {
```

Comment:
I don't understand what you mean by an extra layer. Are you suggesting to use ControlObject rather than ControlProxy?

Reply:
>  verry cheap write only version.

I can't think of a use case for this. Do we have any examples in the current skin system?

Reply:
Writing a value from a GUI without having any representation of it visible to the user doesn't make much sense.

Reply:
Current the date is send from the CO to the proxy to the proxy qml. We need a solution that removes the extra layer.

You are probably right that we don't need a write only version in the qml context.

So please cemove the controlProxy from the controlProxyQml.
This should be easy, by just manualy inline the functions. You will see that almost all will fade away and the resulting obect becomes thin.

If we know how qml connects, it can be optimized even more.

Reply:
Happy to do this in https://github.com/mixxxdj/mixxx/pull/3638 @Be-ing?

Would be interested to know a bit more about when one *would* want to use `ControlProxy` instead @daschuer? In `controlproxy.h` it states:
> It should be used for new code to avoid unnecessary locking during send if no slot is connected.

Doesn't this apply to QML? Or should `ControlProxyQML` be changed to use `ControlObject` directly (inlining the proxy functions as you say), but also ensure to connect to all CO's slots?

Reply:
@jusko if you want to work on this, please go ahead.